src/lib/scheduler/abstract_task.cpp

```diff
@@ -81,20 +90,48 @@ void AbstractTask::join() {
 void AbstractTask::_join_without_replacement_worker() {
   std::unique_lock<std::mutex> lock(_done_mutex);
   _done_condition_variable.wait(lock, [&]() { return _done; });
+
+  // An exception was thrown in the task. We caught it there so that the worker does not crash. This might be an
+  // exception from a predecessor task. If you get an exception here, a good idea is to disable the scheduler so
+  // that it throws immediately. Also, you could use gcc with "catch throw"
+  if (_exception) std::rethrow_exception(_exception);
 }
 
 void AbstractTask::execute() {
   DebugAssert(!(_started.exchange(true)), "Possible bug: Trying to execute the same task twice");
   DebugAssert(is_ready(), "Task must not be executed before its dependencies are done");
 
-  _on_execute();
+  try {
+    _on_execute();
+  } catch (InvalidInputException &e) {
+    if (_exception_callback) {
+      // An exception callback was explicitly provided so we let the receiver handle the exception
+      _exception_callback(std::current_exception());
```

Comment:
mark_as_done() is not called here, why? Someone might be waiting for the task and will now lock forever.